### PR TITLE
Add .equals() method to Material property

### PR DESCRIPTION
**Pull request by MrMagicPenguin**
Add method for [`Material`](https://gltf-transform.donmccurdy.com/classes/core.material.html) to compare material instances. Useful for checking duplicate material; planned as first step for #398.

### CODE REVIEW

**Comment by donmccurdy** [comment]
Thank you! I've merged this in https://github.com/donmccurdy/glTF-Transform/commit/e27344afdb77a160eac0961d8cb39e47e8292259, and made a few adjustments in https://github.com/donmccurdy/glTF-Transform/commit/7db0ffd287c7b58a94dd39c9aa189a99c085c5ea. For the other PR (#407) you may need to resolve conflicts or rebase the branch to get synced with the latest 'main' branch.

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,7 @@ node_modules
 .nyc_output
 .rpt2_cache
 .vscode
+.idea
```
Comment (donmccurdy) [inline .gitignore:10]: I omitted this line, but FWIW you can put this into a `.gitignore` file in your user directory (e.g. `~/.gitignore`) and it should apply to all projects you work on. This makes it easier to contribute to multiple projects without having to customize `.gitignore` for each.

Comment (MrMagicPenguin) [reply .gitignore:10]: Hey thanks for the tip!! I should have simply omitted this change, but I will keep that in mind for the future. Cheers!

**Comment by z1z2** [comment]
@donmccurdy If one material has baseColorTexture and another hasn't, and other properties of the two materials are the same, the equals method will still return true.

```ts
		if (this.getBaseColorTexture() && other.getBaseColorTexture()) {
			if (this.getBaseColorTexture() !== other.getBaseColorTexture()) return false;
			if (!this.getBaseColorTextureInfo()!.equals(other.getBaseColorTextureInfo()!)) {
				return false;
			}
		}
```

**Pull request closed without merging**